eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,240 @@
+  $ChangelogPath = $pkgProperties.ChangeLogPath
+  
+  if (!(Test-Path $ChangelogPath)) {
+    LogError "Changelog path [$ChangelogPath] is invalid."
+    exit 1
+  }
+  
+  $ReleaseStatus = "$(Get-Date -Format $CHANGELOG_DATE_FORMAT)"
+  $ReleaseStatus = "($ReleaseStatus)"
+  $ChangeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $ChangelogPath
+  LogDebug "Adding new ChangeLog entry for Version [$Version]"
+  $Content = @()
+  $Content += ""
+  $Content += "### Dependency Updates"
+  $Content += ""
+  $Content += "Upgraded ``azure-core`` and other dependencies for the library."
+  $Content += ""
+  $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus -Content $Content
+  if ($newChangeLogEntry) {
+    $ChangeLogEntries.Insert(0, $Version, $newChangeLogEntry)
+  }
+  else {
+    LogError "Failed to create new changelog entry"
+    exit 1
+    }
+  Set-ChangeLogContent -ChangeLogLocation $ChangelogPath -ChangeLogEntries $ChangeLogEntries
+}
+
+if('' -eq $PatchVersion) {
+  $PatchVersion = GetPatchVersion -ReleaseVersion $ReleaseVersion
+  if('' -eq $PatchVersion) {
+    LogError "Could not fetch the patch version. Exitting ..."
+    exit
+  }
+}
+Write-Information "PatchVersion is: $PatchVersion"
+
+$RemoteName = GetRemoteName -MainRemoteUrl $MainRemoteUrl
+if($null -eq $RemoteName) {
+    LogError "Could not fetch the remote name for the URL $MainRemoteUrl Exitting ..."
+    exit
+}
+Write-Information "RemoteName is: $RemoteName"
+
+if('' -eq $BranchName) {
+  $ArtifactNameToLower = $ArtifactName.ToLower()
+  $BranchName = "release/$ArtifactNameToLower/$PatchVersion"
+}
+Write-Information "BranchName is: $BranchName"
+
+try {
+  Write-Information "Reseting the $ArtifactName sources to the release $ReleaseTag."
+  ## Creating a new branch
+  $CmdOutput = git checkout -b $BranchName $RemoteName/main
+  if($LASTEXITCODE -ne 0) {
+    LogError "Could not checkout branch $BranchName, please check if it already exists and delete as necessary. Exitting..."
+    exit
+  }
+  
+  ## Hard reseting it to the contents of the release tag.
+  ## Fetching all the tags from the remote branch
+  Write-Information "Fetching all the tags from $RemoteName"
+  $CmdOutput = git fetch $RemoteName --tags
+  
+  $cmdOutput = git restore --source $ReleaseTag -W -S $ArtifactDirPath
+  if($LASTEXITCODE -ne 0) {
+    LogError "Could not restore the changes for release tag $ReleaseTag"
+    exit
+  }
+
+  if(Test-Path $TestResourcesFilePath) {
+    $cmdOutput = git restore --source $ReleaseTag -W -S $TestResourcesFilePath
+  }
+  
+ 
+  if(Test-Path $CheckStyleSuppressionFilePath) {
+    $cmdOutput = git restore --source $ReleaseTag -W -S $CheckStyleSuppressionFilePath
+  }
+  
+  if(Test-Path $SpotBugsFilePath) {
+    $cmdOutput = git restore --source $ReleaseTag -W -S $SpotBugsFilePath
+  }
+
+  ## Commit these changes.
+  $cmdOutput = git commit -a -m "Reset changes to the patch version."
```

Comment:
I suggest doing a `git add -A` instead of passing `-a` to commit as the `-A` will help ensure things like renamed, added or deleted files also get picked up correctly.

Reply:
Ooh- TIL. Thanks!